Bug/fix scroll on Project Health Score click

divyanshu-vr opened the pull request:
<!-- Thanks for contributing to OWASP Nest!-->

## Proposed change
Fixed scroll on Project Health Score click

<!-- Don't forget to link your PR to an existing issue if any.-->
Resolves #2034 

<!-- Describe the big picture of your changes.-->
Fixed the bug and added tests.

Feature- we can add a feature where pressing Enter or Spacebar on the badge scrolls down. I've optimized it for so, if we ever need to do it in the future.

## Checklist

- [ ] I've read and followed the [contributing guidelines](https://github.com/OWASP/Nest/blob/main/CONTRIBUTING.md).
- [x] I've run `make check-test` locally; all checks and tests passed.

divyanshu-vr commented on frontend/src/components/MetricsScoreCircle.tsx:
```diff
@@ -1,18 +1,33 @@
   return (
     <Tooltip content={'Current Project Health Score'} placement="top">
+      {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
       <div
-        className={`group relative flex h-14 w-14 flex-col items-center justify-center rounded-full shadow-md transition-all duration-300 hover:scale-105 hover:shadow-lg ${scoreStyle}`}
+        className={`group relative flex h-14 w-14 flex-col items-center justify-center rounded-full shadow-md transition-all duration-300 hover:scale-105 hover:shadow-lg ${scoreStyle} ${
+          onClick ? 'cursor-pointer' : ''
+        }`}
+        onClick={handleClick}
```
we can leave this for a future update to add functionality for keyboard.

divyanshu-vr commented:
Hi @kasya, @arkid15r,
I’ve fixed the bug and also left scope to add keyboard handler functionality in the future. Here’s the video attached below showing the implementation.


https://github.com/user-attachments/assets/4dae5c57-0322-43c1-ba6e-ad22966c903d

kasya reviewed (changes requested):
@divyanshu-vr Hi! There are some conflicts with your current changes. Please resolve those so I could review the PR 👍🏼

kasya reviewed (changes requested):
This works! I did left some comments about implementation though and also you need to update your PR - you removed a bunch of stuff that we still use in the `CardDetailsPage` component.

divyanshu-vr commented on frontend/src/components/CardDetailsPage.tsx:
```diff
@@ -10,12 +10,9 @@ import {
-import Link from 'next/link'
-import { useRouter } from 'next/navigation'
-import { useSession } from 'next-auth/react'
-import type { ExtendedSession } from 'types/auth'
```
I'll revert back to how it was before.

divyanshu-vr commented on frontend/src/components/CardDetailsPage.tsx:
```diff
@@ -96,9 +93,15 @@ const DetailsCard = ({
-              <Link href="#issues-trend">
-                <MetricsScoreCircle score={healthMetricsData[0].score} />
-              </Link>
+              <a
```
> I don't think we use regular anchor tag anywhere else. Is there a way to make it work with `Link` for consistency? Because it seems that the main issue was with incorrect offset, right?

hi @kasya, I’ve replaced the anchor tag with Link and it works fine now. ill send the push soon.

Also, do you think utils\scrollToAnchor.ts is unnecessary?
From my perspective, the incorrect offset issue was happening because the scroll didn’t account for the header/nav height — the offset itself was somewhat fine, but the content was getting hidden under the header.

If you prefer, I can take a simpler approach and redo this entirely. In that case, we can close this PR and I’ll start a fresh one.

kasya reviewed (approved):
@divyanshu-vr Thanks for working on this!

I refactored MetricsScoreCircle to handle click logic internally instead of wrapping with `Link` components. Also added onClick and clickable props for flexible usage, improved accessibility with proper button roles and keyboard navigation. This provides cleaner architecture and better separation of concerns.

divyanshu-vr commented:
> @divyanshu-vr Thanks for working on this!
> 
> I refactored MetricsScoreCircle to handle click logic internally instead of wrapping with `Link` components. Also added onClick and clickable props for flexible usage, improved accessibility with proper button roles and keyboard navigation. This provides cleaner architecture and better separation of concerns.

These changes make it even better — thank you, @kasya! I had implemented keyboard navigation initially but later removed it, as I was asked on the issue page to leave it for a future enhancement. Do you have any specific feedback for me? I’m new to OSS and still learning best practices.
